std/concurrency.d

```diff
@@ -1682,13 +1683,54 @@ class Generator(T) :
 
 
     /**
-     * Returns the most recently generated value.
+     * Returns the most recently generated value by shallow copy.
      */
     final T front() @property
     {
         return *m_value;
     }
 
+    /**
+     * Returns the most recently generated value without excuting a copy
+     * contructor. Will not compile for element types defining a
+     * postblit, because Generator does not return by reference.
+     */
+    final T moveFront()
+    {
+        import std.algorithm, std.range;
+        static if (!hasElaborateCopyConstructor!T)
+        {
+            return front;
+        }
+        else
+        {
+            static assert(0,
+                    "Fiber front is rvalue and thus cannot be moved when it defines a postblit.");
+        }
+    }
+
+    final int opApply(scope int delegate(T) loopBody)
+    {
+        int broken;
+        for(; !empty; popFront())
+        {
+            broken = loopBody(front);
+            if(broken) break;
+        }
+        return broken;
+    }
+
+    final int opApply(scope int delegate(size_t, T) loopBody)
```

Comment:
Hmm if we have to add `opApply` to everything that is an `InputRange` the range concept of "building block" would have somehow failed. Do you know about `enumerate` and why doesn't it work for you?